docs/source/torch.compiler.rst

```diff
@@ -0,0 +1,116 @@
+.. _torch.compiler_overview:
+
+torch.compiler
+==============
+
+``torch.compiler`` is a namespace through which some of the internal compiler
+methods are surfaced for user consumption. The main function and the feature in
+this namespace is ``torch.compile``.
+
+``torch.compile`` is a PyTorch function introduced in PyTorch 2.x that aims to
+solve the problem of accurate graph capturing in PyTorch and ultimately enable
+software engineers to run their PyTorch programs faster. ``torch.compile`` is
+written in PyThon and it marks the transition of PyTorch from C++ to PyThon.
+
+``torch.compile`` leverages the following underlying technologies:
+
+* TorchDynamo (``torch._dynamo``) is an internal API that uses a CPython
+  feature called the Frame Evaluation API to safely capture PyTorch graphs.
+  Methods that are available externally for PyTorch users are surfaced
+  through the ``torch.compiler`` namespace.
+
+* TorchInductor is the default ``torch.compile`` deep learning compiler
+  that generates fast code for multiple accelerators and backends. You
+  need to use a backend compiler to make speedups through ``torch.compile``
+  possible. For NVIDIA and AMD GPUs, it leverages OpenAI Triton as the key
+  building block.
+
+* AOTAutograd captures not only the user-level code, but also backpropagation,
+  which results in capturing the backwards pass "ahead-of-time". This enables
+  acceleration of both forwards and backwards pass using TorchInductor.
+
+.. note:: In some cases, the terms ``torch.compile``, TorchDynamo, ``torch.compiler``
+  might be used interchangeably in this documentation.
+
+As mentioned above, to run your workflows faster, ``torch.compile`` through
+TorchDynamo requires a backend that converts the captured graphs into fast
+machine code. Different backends can result in various optimization gains.
+The default backend is called TorchInductor, also known as *inductor*,
+TorchDynamo has a growing list of backends developed by our partners,
+which can be found in the `backends <https://github.com/pytorch/pytorch/tree/main/torch/_dynamo/backends>`_
```

Comment:
should be torch.compiler.list_backends

Also we actually don't want this list to grow XD so would rather we change verbiage to we're not accepting new backends but you can register them out of core following links for vendors